sci-physics/thepeg/files/thepeg-2.3.0-javac_source.patch

```diff
@@ -0,0 +1,17 @@
+We are changing the javac -soure value from 1.4 to 1.8 which is supported by all
+Java versions presently available in ::gentoo.
+
+TODO:
+Replace '1.8' with a variable to be set with $(java-pkg_get-source) in the ebuild.
```

Comment:
Maybe set `JAVA_PKG_GET_SOURCE` in the configure.ac and `sed` it?

Reply:
I think so, but I can try this later today:

```
THEPEG_HAS_JAVA([JAVA_PKG_GET_SOURCE], [], [with_javagui=no; AC_MSG_NOTICE([Java GUI disabled])])
```

and early in `src_prepare`
```
src_prepare() {
     sed -i "s/JAVA_PKG_GET_SOURCE/$(java-pkg_get-source)/g" configure.ac
```

Reply:
You are the (proxied) maintainer. I'll leave that to you. 
But what about removing versions [2.2.3-r1](https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-physics/thepeg/thepeg-2.2.3-r1.ebuild) and [2.2.2-r1](https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-physics/thepeg/thepeg-2.2.2-r1.ebuild)? Or do you prefer backporting all changes to them?

Reply:
Given that only https://github.com/gentoo/sci/blob/master/sci-physics/herwig/herwig-7.3.0.ebuild depends on `thepeg` I don't mind dropping them (after checking that current `herwig` works)

Reply:
> Maybe set `JAVA_PKG_GET_SOURCE` instead of `1.8` in the configure.ac and `sed` it?

One more thing (could be done in a follow-up PR): It should also have `-target $(java-pkg_get-target)` in the `configure` file. I just discovered this by emerging with `JAVA_PKG_STRICT=true` in `/etc/portage/make.conf` and `javatoolkit` installed.